tests/unit/test_hookutils.py

```diff
@@ -322,3 +323,35 @@ def test_collect_data_all_included(data_lists):
 def test_get_module_file_attribute_non_exist_module():
     with pytest.raises(ImportError):
         get_module_file_attribute('pyinst_nonexisting_module_name')
+
+
+@pytest.mark.parametrize("egg_path,name,target", [
+    # Something installed via `pip install -e .`.
+    ("editable/install/CodeChat.egg-info", "CodeChat", "CodeChat.egg-info"),
+    # An egg distribution - it's unlikely we'll ever see these now.
+    ("lib/site-packages/pypubsub-3.3.0-py2.7.egg/EGG-INFO",
+     "pypubsub", "pypubsub-3.3.0-py2.7.egg/EGG-INFO"),
+    # A classic wheel-installed distribution.
+    ("lib/site-packages/zest.releaser-6.2.dist-info",
+     "zest.releaser", "zest.releaser-6.2.dist-info"),
+    # Must be tolerant to case and -/_ mismatch.
+    ("/site-packages/importlib_metadata-4.0.1.dist-info",
+     "ImPorTlib-mEtADatA", "importlib_metadata-4.0.1.dist-info")
+])
+def test_copy_metadata_dest(egg_path, name, target):
+    """Test choosing dest path for copy_metadata() across distribution types.
+    """
+    # Convert posix style filenames to native paths. i.e. replace '/' with '\'
+    # on Windows.
+    egg_path = str(pathlib.Path(*pathlib.PurePosixPath(egg_path).parts))
+    target = str(pathlib.Path(*pathlib.PurePosixPath(target).parts))
```

Comment:
Wouldn't passing `PurePosixPath` to `PurePath` constructor do the trick without having to explicitly break path into parts? 
I.e.,  `egg_path = str(pathlib.PurePath(pathlib.PurePosixPath(egg_path)))`

Reply:
For the given paths, we might actually get away without explicit construction of `PurePosixPath` as well...

Reply:
Huh it does as well. I thought I'd tried both of those already. 
```python
>>> str(pathlib.PureWindowsPath(pathlib.PurePosixPath(r"foo/bar")))
'foo\\bar'
>>> str(pathlib.PureWindowsPath(r"foo/bar"))
'foo\\bar'
```

Reply:
How about `egg_path.as_posix()`?

Reply:
Ah sorry, I didn't see the new commit, my bad :)